src/main/java/gift/Service/OptionKeeperService.java

```diff
@@ -0,0 +1,35 @@
+package gift.Service;
+
+import gift.Exception.OptionValidationException;
+import gift.Model.Option;
+import gift.Model.Product;
+import gift.Repository.OptionRepository;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class OptionKeeperService {
+    private final OptionRepository optionRepository;
+
+    public OptionKeeperService(OptionRepository optionRepository){
+        this.optionRepository = optionRepository;
+    }
+
+    public void checkUniqueOptionName(Product product, String name){
+        optionRepository.findByProduct(product)
+                .stream()
+                .filter(it->it.getName().equals(name))
+                .findFirst()
+                .ifPresent(it -> { // 옵션이 존재할 경우
+                    throw new DataIntegrityViolationException("이미 해당 상품내에 동일한 이름의 옵션이 존재합니다");
+                });
+    }
+
+    public void checkHasAtLeastOneOption(Product product) {
+        List<Option> optionList = optionRepository.findByProduct(product);
+        if (optionList.size() == 1 )
```

Comment:
```suggestion
        if (optionList.size() < 1 )
```
optionList의 size가 0이면 어떻게 될까요?

Reply:
저도 안그래도 특정 상품에 대한 옵션이 최소한 1개 있어야 되므로 <= 1로 해야되나 라는 생각을 코드 짤 당시에 고민을 하였습니다. 하지만 상품을 추가할 때 무조건 옵션을 하나 추가해야 상품이 등록이 되기에 상품 객체 생성 시 옵션 갯수의 기본값이 1로 시작이 되므로 == 1로 작성해도 무방하다고 생각이 들었습니다. 그래서 제 생각은 굳이 수정할 필요는 없어 보인다고 느껴지는데 혹시 제가 미처 고려하지 못하는 예외적인 상황이 있을까요?

Reply:
넵!! 다시보니까 그렇겠네요. 잘 작성해주셨군요 😅